File: API/WorkoutTracker.API/Models/ExecutedWorkoutDTO.cs
using System;
using System.Collections.Generic;

namespace WorkoutTracker.API.Models
{
    public record ExecutedWorkoutDTO(Guid Id, DateTime CreatedDateTime, DateTime? ModifiedDateTime, string Name,
        Guid WorkoutId,
        DateTime? StartDateTime,
        DateTime? EndDateTime,
        string Journal,
        int Rating,
        IEnumerable<ExecutedExerciseDTO> Exercises) 
        : NamedEntityDTO(Id, CreatedDateTime, ModifiedDateTime, Name);
}

File: workout-tracker/src/app/workouts/models/executed-workout-dto.ts
import { NamedEntityDTO } from "./named-entity-dto";
import { ExecutedExerciseDTO } from "./executed-exercise-dto";

export class ExecutedWorkoutDTO extends NamedEntityDTO {
  public workoutId: string = '';
  public startDateTime: Date | null = null;
  public endDateTime: Date | null = null;
  public journal: string | null = null;
  public rating: number = 0;
  public exercises: Array<ExecutedExerciseDTO> = [];
}
